### PR TITLE
Enabling Multi Device support for UEP

### PR DESCRIPTION
**Pull request by MaajidKhan**
**Description**: 
Added Multi Device plugin support for UEP

**Motivation and Context**
Multi-Device plugin automatically assigns inference requests to available computational devices to execute the requests in parallel. Potential gains are as follows

-> Improved throughput that multiple devices can deliver (compared to single-device execution)
-> More consistent performance, since the devices can now share the inference burden (so that if one device is becoming too 
     busy, another device can take more of the load)

### CODE REVIEW

File: onnxruntime/core/providers/openvino/ov_versions/capability_2021_1.cc [hunk 1]
```diff
@@ -116,6 +116,7 @@ bool IsOpSupported(std::string name, std::string device) {
       "Sub",
       "Sum",
       "Tanh",
+      "Tile",
```
Comment (MaajidKhan) [inline onnxruntime/core/providers/openvino/ov_versions/capability_2021_1.cc:119]: Have added this op for some experiment purpose by mistake, will remove it later

File: onnxruntime/core/providers/openvino/ov_versions/capability_2021_1.cc [hunk 2]
```diff
@@ -163,6 +164,7 @@ bool IsOpSupported(std::string name, std::string device) {
     "Tan",
   };
   std::set<std::string> supported_ops_vpu = {
+    "ArgMax",
```
Comment (MaajidKhan) [inline onnxruntime/core/providers/openvino/ov_versions/capability_2021_1.cc:167]: Added this op for VPU to fully support MobilenetV2-SSDLite model on UEP and with this fix, the model is working on UEP with OpenVINO Latest master

File: BUILD.md [hunk 3]
```diff
@@ -396,16 +396,18 @@ See more information on the OpenVINO Execution Provider [here](./docs/execution_
 | <code>VAD-M_FP16</code> | Intel<sup>®</sup> Vision Accelerator Design based on 8 Movidius<sup>TM</sup> MyriadX VPUs |
 | <code>VAD-F_FP32</code> | Intel<sup>®</sup> Vision Accelerator Design with an Intel<sup>®</sup> Arria<sup>®</sup> 10 FPGA |
 | <code>HETERO:<DEVICE_TYPE_1>,<DEVICE_TYPE_2>,<DEVICE_TYPE_3>...</code> | All Intel<sup>®</sup> silicons mentioned above |
+| <code>MULTI:<DEVICE_TYPE_1>,<DEVICE_TYPE_2>,<DEVICE_TYPE_3>...</code> | All Intel<sup>®</sup> silicons mentioned above |
```
Comment (smkarlap) [inline BUILD.md:399]: Could you add a summary description of Hetero and Multi plugins beyond supporting all silicons?

File: onnxruntime/core/providers/openvino/ov_versions/capability_2021_1.cc [hunk 4]
```diff
@@ -208,6 +210,35 @@ bool IsOpSupported(std::string name, std::string device) {
...
     }
+  } else if (device.find("MULTI") == 0) {
+    std::vector<std::string> devices;
+    std::stringstream s_stream(device);
+    while(s_stream.good()) {
+      std::string substr;
+      getline(s_stream, substr, ',');
+      devices.push_back(substr);
+    }
+    if (common_supported_ops.count(name)) {
```
Comment (smkarlap) [inline onnxruntime/core/providers/openvino/ov_versions/capability_2021_1.cc:221]: the count() function of std::set, which is the type of common_supported_ops supported_ops_*pu, returns a value of type size_type which is an unsigned integer. It is not a clean practice to use it as an expression of if,else statements which need to be boolean. Use expressions like "common_supported_ops.count(name) == 0" instead

File: onnxruntime/core/providers/openvino/ov_versions/capability_2021_1.cc [hunk 5]
```diff
@@ -208,6 +210,35 @@ bool IsOpSupported(std::string name, std::string device) {
...
+      return true;
+    }
+    for (auto& it : devices) {
+      if(it == "MYRIAD" || "HDDL") {
+        if (!supported_ops_vpu.count(name))  {
+          return false;
+        }
+      }
+      if(it == "GPU") {
+       if (!supported_ops_gpu.count(name))  {
```
Comment (smkarlap) [inline onnxruntime/core/providers/openvino/ov_versions/capability_2021_1.cc:231]: same comment as above

File: onnxruntime/core/providers/openvino/ov_versions/capability_2021_1.cc [hunk 6]
```diff
@@ -208,6 +210,35 @@ bool IsOpSupported(std::string name, std::string device) {
...
+      std::string substr;
+      getline(s_stream, substr, ',');
+      devices.push_back(substr);
+    }
+    if (common_supported_ops.count(name)) {
+      return true;
+    }
+    for (auto& it : devices) {
+      if(it == "MYRIAD" || "HDDL") {
+        if (!supported_ops_vpu.count(name))  {
```
Comment (smkarlap) [inline onnxruntime/core/providers/openvino/ov_versions/capability_2021_1.cc:226]: same comment as above

File: onnxruntime/core/providers/openvino/ov_versions/capability_2021_1.cc [hunk 7]
```diff
@@ -208,6 +210,35 @@ bool IsOpSupported(std::string name, std::string device) {
...
+          return false;
+        }
+      }
+      if(it == "GPU") {
+       if (!supported_ops_gpu.count(name))  {
+          return false;
+        }
+      }
+      if(it == "CPU") {
+        if (!supported_ops_cpu.count(name))  {
```
Comment (smkarlap) [inline onnxruntime/core/providers/openvino/ov_versions/capability_2021_1.cc:236]: same comment as above

File: onnxruntime/core/providers/openvino/ov_versions/capability_2021_1.cc (hunk 4 above)
Comment (MaajidKhan) [reply onnxruntime/core/providers/openvino/ov_versions/capability_2021_1.cc:221]: Done @smkarlap

File: onnxruntime/core/providers/openvino/ov_versions/capability_2021_1.cc (hunk 5 above)
Comment (MaajidKhan) [reply onnxruntime/core/providers/openvino/ov_versions/capability_2021_1.cc:231]: Done

File: onnxruntime/core/providers/openvino/ov_versions/capability_2021_1.cc (hunk 6 above)
Comment (MaajidKhan) [reply onnxruntime/core/providers/openvino/ov_versions/capability_2021_1.cc:226]: Done

File: onnxruntime/core/providers/openvino/ov_versions/capability_2021_1.cc (hunk 7 above)
Comment (MaajidKhan) [reply onnxruntime/core/providers/openvino/ov_versions/capability_2021_1.cc:236]: Done

File: BUILD.md (hunk 3 above)
Comment (MaajidKhan) [reply BUILD.md:399]: @smkarlap In the build.md file (https://github.com/intel/onnxruntime/blob/maajid/implemet_multi_device/BUILD.md#build-instructions-6),
I have added how Hetero Device type or MULTI Device type target can be added and with some examples.

The summary of Hetero and Multi plugins is already added here (https://github.com/intel/onnxruntime/blob/maajid/implemet_multi_device/docs/execution_providers/OpenVINO-ExecutionProvider.md#heterogeneous-execution-for-openvino-ep)

**Pull request closed without merging**